pinot-spi/src/main/java/org/apache/pinot/spi/data/function/evaluators/GroovyExpressionEvaluator.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.data.function.evaluators;
+
+import com.google.common.base.Splitter;
+import groovy.lang.Binding;
+import groovy.lang.GroovyShell;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * An {@link ExpressionEvaluator} for evaluating schema transform expressions written in Groovy.
+ * GroovyShell is used to execute expressions.
+ *
+ * The transform expression must follow the convention Groovy({expression}, arguments1, argument2...)
+ * For example:
+ * "dimensionFieldSpecs": [
+ *     {
+ *       "name": "fullName",
+ *       "dataType": "STRING",
+ *       "transformFunction": "Groovy({firstName+' '+lastName}, firstName, lastName)"
+ *     }
+ *  ]
+ */
+public class GroovyExpressionEvaluator implements ExpressionEvaluator {
+
+  private static final String GROOVY_EXPRESSION_PREFIX = "Groovy";
+  private static final String GROOVY_FUNCTION_REGEX = "Groovy\\(\\{(?<script>.+)}(,(?<arguments>.+))?\\)";
```

Comment:
Is this the only syntax available for a groovy script?

Reply:
This is the convention we will follow
`FunctionType({script}, argument1, argument2...)`
For Groovy that becomes
`Groovy({groovy script}, argument1, argument2)`

This can be easily extended to other function types. I've put some more details in the PR description, and also in the design doc linked in the PR description